lib/measurable/interfaces.rb

```diff
@@ -0,0 +1,52 @@
+module Measurable
+  module Interface
+    class DistanceByName
+      def initialize(name)
+        @name = name
+      end
+
+      def distance(a, b)
+        Measurable.send(@name, a, b)
+      end
+    end
+    def getObject(measure)
+      DistanceByName.new(measure)
+    end
+  end
+
+  extend Measurable::Interface
+
+  module MeasurableObject
+    # Mixin to add ability for any object,
+    # which implements coords,
+    # to calc distance with 'calc_distance' function
+    def distance(other, measure = nil, &distance)
```

Comment:
Well, that comment probably needs fixing, it seems the correct name is `distance`, not `calc_distance`.

In Ruby, we usually prefer longer, complete names instead of abbreviations. So it would be `coordinates` instead of `coords`.